File: content/DP/lab3/src/main/java/ru/prokdo/model/file/FileReader.java
package ru.prokdo.model.file;

import java.nio.file.Paths;
import java.io.BufferedReader;
import java.nio.file.Files;


public final class FileReader {
    public final String filePath;

    public FileReader(String filePath) {
        if (filePath == null) throw new IllegalArgumentException("File path cannot be null");
        
        this.filePath = filePath;
    }

    public String read() {
        try { 
            BufferedReader reader = Files.newBufferedReader(Paths.get(filePath)); 

            String result = reader.readLine();

            reader.close();

            return result;
        }
        catch (Exception exception) { return null; }
    }
}

File: content/DP/lab3/src/main/kotlin/model/file/FileReader.kt
package model.file

import java.io.File

object FileReader {
    fun read(file: File): List<String>? {
        val result = mutableListOf<String>()
        try { file.forEachLine { result.add(it) } }
        catch (ignored: Exception) { return null }

        return result.toList()
    }
}